tests/unit/s2n_connection_test.c

```diff
@@ -162,5 +163,64 @@ int main(int argc, char **argv)
         EXPECT_SUCCESS(s2n_connection_free(server_conn));
     }
 
+    /* Test: get selected digest alg */
+    {
+        struct s2n_connection *conn;
+        EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_CLIENT));
+
+        s2n_handshake_hash_algorithm output = { 0 };
+        EXPECT_FAILURE(s2n_connection_get_selected_digest_algorithm(conn, &output));
+        EXPECT_FAILURE(s2n_connection_get_selected_client_cert_digest_algorithm(conn, &output));
+
+        s2n_hash_algorithm inputs[] = { S2N_HASH_MD5, S2N_HASH_SHA1, S2N_HASH_SHA224, S2N_HASH_SHA256, S2N_HASH_SHA384, 
+                                        S2N_HASH_SHA512, S2N_HASH_MD5_SHA1 };
+        s2n_handshake_hash_algorithm expected_output[] = { S2N_HANDSHAKE_HASH_MD5, S2N_HANDSHAKE_HASH_SHA1, 
+                                                           S2N_HANDSHAKE_HASH_SHA224, S2N_HANDSHAKE_HASH_SHA256, 
+                                                           S2N_HANDSHAKE_HASH_SHA384, S2N_HANDSHAKE_HASH_SHA512, 
+                                                           S2N_HANDSHAKE_HASH_MD5_SHA1 };
+
+        for (int i = 0; i < s2n_array_len(inputs); i++) {
+            conn->secure.client_cert_sig_scheme.hash_alg = inputs[i];
+            EXPECT_SUCCESS(s2n_connection_get_selected_client_cert_digest_algorithm(conn, &output));
+            conn->secure.client_cert_sig_scheme.hash_alg = 0;
```

Comment:
How does this behave if we're not doing client auth? It's an optional feature

Reply:
Do you think that's the correct behavior?

Reply:
It's easier to understand if the invalid state is enumerated so I will add a case for that in the public enum, instead of returning an error.

Reply:
I think I would prefer adding a HANDSHAKE_HASH_NONE case over erroring, since it's not really an invalid case.

Reply:
Renamed to your original suggestion after reading https://tools.ietf.org/html/rfc5246#section-7.4.1.4.1